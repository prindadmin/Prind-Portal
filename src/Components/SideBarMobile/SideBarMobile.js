import React, { Component } from 'react'
import ListItem from '../Common/page-tile'
import ItemIcon from '../Common/ItemIcon'

import { Button, MenuItem } from "@blueprintjs/core";
import { Select } from "@blueprintjs/select";

import PAGENAMES from '../../Data/pageNames'

// TODO: work out why the popover arrow isn't pointing to the menu button

export class SideBar extends Component {

  handleElementSelect = (query) => {
    console.log(query)
    console.log('linkTo: ', query.linkTo);

    const { history, projects } = this.props
    const { projectId } = projects.chosenProject
    var projectRoute = ''

    // Get the current project (if present)
    if(projectId !== "") {
      projectRoute = `/${projectId}`
    }

    history.push(`${query.linkTo}${projectRoute}`)
  }

  // Render each item in the list
  itemRenderer = (item) => {
    return (
      <MenuItem
        key={item.name}
        onClick={(e) => this.handleElementSelect(item)}
        text={item.name}
        shouldDismissPopover
      />
    )
  }

  pageNamesToList = () => {

    const { chosenProject } = this.props.projects
    const { projectId, projectType } = chosenProject

    const menuEntries = PAGENAMES[projectType] === undefined ? PAGENAMES["CDM2015Project"] : PAGENAMES[projectType]

    const pageNameList = Object.keys(menuEntries).map(page => {
      return {
        name: menuEntries[page].name,
        linkTo: menuEntries[page].linkTo
      }
    })

    return pageNameList
  }


  render() {

    const { projectId } = this.props.projects.chosenProject
    const pathForProject = projectId !== "" ? `/${projectId}` : ""

    return (
        <div className='chooser-section col-12'>
          <div className='sidebar'>
            <Select
              items={this.pageNamesToList()}
              itemRenderer={this.itemRenderer}
              filterable={false}
              >
              <Button icon={<ItemIcon size='1x' type='burger' />} />
            </Select>
          </div>
        </div>
    )
  }
}


export default SideBar
