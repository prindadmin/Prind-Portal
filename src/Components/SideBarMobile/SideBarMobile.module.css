.chooserContainer {
  width: 30px;
  height: 30px;
  position: absolute;
  top: calc(var(--header-height) + 0.33em);
  left: 0.33em;
  z-index: 100;
}
