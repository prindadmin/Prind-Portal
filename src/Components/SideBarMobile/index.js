import SideBarMobile from './SideBarMobileContainer'

export default SideBarMobile
