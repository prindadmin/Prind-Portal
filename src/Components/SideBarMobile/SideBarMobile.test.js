import React from 'react';
import { shallow } from 'enzyme';

import SideBarMobile from './SideBarMobile'

beforeEach(() => {
  process.env = Object.assign(process.env, {
    REACT_APP_PORTAL: 'CDM2015Project'
  });
});

it('Should render CDM project Sidebar', () => {
  process.env = Object.assign(process.env, {
    REACT_APP_PORTAL: 'CDM2015Project'
  });
  const props = {
    projects: {
      chosenProject: {
        projectId: '1234',
        projectType: "CDM2015Project"
      }
    },
    location: {
      pathname: "/occupation/TestFromForm3Dhsf2021-01-11"
    }
  };
  const component = shallow(<SideBarMobile {...props} />);
  expect(component).toMatchSnapshot();
});


it('Should render DSHF project Sidebar', () => {
  process.env = Object.assign(process.env, {
    REACT_APP_PORTAL: 'DHSFProject'
  });
  const props = {
    projects: {
      chosenProject: {
        projectId: '1234',
        projectType: "DHSFProject"
      }
    },
    location: {
      pathname: "/occupation/TestFromForm3Dhsf2021-01-11"
    }
  };
  const component = shallow(<SideBarMobile {...props} />);
  expect(component).toMatchSnapshot();
});
