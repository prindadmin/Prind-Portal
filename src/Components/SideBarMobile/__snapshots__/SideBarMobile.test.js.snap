// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CDM project Sidebar 1`] = `
<div
  className="chooser-section col-12"
>
  <div
    className="sidebar"
  >
    <Blueprint3.Select
      filterable={false}
      itemRenderer={[Function]}
      items={
        Array [
          Object {
            "linkTo": "/project",
            "name": "Project Details",
          },
          Object {
            "linkTo": "/team",
            "name": "Project Team",
          },
          Object {
            "linkTo": "/inception",
            "name": "Inception",
          },
          Object {
            "linkTo": "/feasibility",
            "name": "Feasibility",
          },
          Object {
            "linkTo": "/design",
            "name": "Design",
          },
          Object {
            "linkTo": "/tender",
            "name": "Tender",
          },
          Object {
            "linkTo": "/construction",
            "name": "Construction",
          },
          Object {
            "linkTo": "/handover",
            "name": "Handover",
          },
          Object {
            "linkTo": "/occupation",
            "name": "Occupation",
          },
          Object {
            "linkTo": "/refurbishment",
            "name": "Refurbishment",
          },
        ]
      }
    >
      <Blueprint3.Button
        icon={
          <ItemIcon
            size="1x"
            type="burger"
          />
        }
      />
    </Blueprint3.Select>
  </div>
</div>
`;

exports[`Should render DSHF project Sidebar 1`] = `
<div
  className="chooser-section col-12"
>
  <div
    className="sidebar"
  >
    <Blueprint3.Select
      filterable={false}
      itemRenderer={[Function]}
      items={
        Array [
          Object {
            "linkTo": "/project",
            "name": "Project Details",
          },
          Object {
            "linkTo": "/team",
            "name": "Project Team",
          },
          Object {
            "linkTo": "/inception",
            "name": "Inception",
          },
          Object {
            "linkTo": "/feasibility",
            "name": "Feasibility",
          },
          Object {
            "linkTo": "/design",
            "name": "Design",
          },
          Object {
            "linkTo": "/tender",
            "name": "Tender",
          },
          Object {
            "linkTo": "/construction",
            "name": "Construction",
          },
          Object {
            "linkTo": "/handover",
            "name": "Handover",
          },
          Object {
            "linkTo": "/occupation",
            "name": "Occupation",
          },
          Object {
            "linkTo": "/refurbishment",
            "name": "Refurbishment",
          },
        ]
      }
    >
      <Blueprint3.Button
        icon={
          <ItemIcon
            size="1x"
            type="burger"
          />
        }
      />
    </Blueprint3.Select>
  </div>
</div>
`;
